### DISCUSSION TITLE
Use Macad3D in commercial application

### DISCUSSION THREAD

**Discussion by borioan** [Macad3D/Macad3D#2] [Q&A] [answered]
Hello!
First of all I would like to thank you for developing this amazing project!!
My main concern is this one: **it is possible to use _parts_ of Macad3D in a commercial application?** 
In the [license page of Open CASCADE Technology](https://old.opencascade.com/doc/occt-7.5.0/overview/html/index.html#intro_license) there are 3 main points to take care before use it in a commercial application:

_1. Add the notice visible to the users of your application clearly stating that Open CASCADE Technology is used in this application, and that they have rights in this regard according to LGPL. Such notice can be added in About dialog box (this is mandatory if this box contains copyright statements) or a similar place and/or in the documentation. The text of LGPL license should be accessible to the user.
2. Make the copy of OCCT sources used by the application available to its users, and if necessary provide instructions on how to build it in a way compatible with the application.
3. Ensure that the user actually can exercise the right to run your application with a modified version of OCCT. If the application is distributed in a form that does not allow the user to modify OCCT part (e.g. the application is linked to OCCT statically or is distributed via AppStore on iOS, GooglePlay on Android, Windows Store, etc.), the application should be provided separately in a modifiable form, with all materials needed for the user to be able to run the application with a modified version of OCCT._

I'm very new to OCCT technology and I need to know if Macad3D use a modified version of OCCT, in order to take into account and respect point 3 from above obligations. If Macad3D use a modified version of OCCT do you have any idea how to use your code to respect these license obligations? Anyway, Macad3D is an excellent OCCT implementation and among the very few of 
it using C#. If there are any chances to use parts of it in commercial application with respecting the licensing conditions I will coming back here for a lot of questions and tests!!
Thank you very much!

**Comment by dipts** [comment, marked as answer]
Thank you for your interest in this project. **Macad|3D uses an unmodified version of OCCT.** The official version, which OpenCASCADE provides for download, can be used to build the project. For the setup version I use the appropriately tagged sources directly from the official OCCT git repo.

Smaller modifications are currently only made to generate the wrapper classes, which are summarized in a patch file. See [Readme.md](https://github.com/Macad3D/Macad3D/blob/main/Readme.md). This patch is not required for the compilation and is not automatically applied during the import. I have corrected the documentation accordingly. The reason for the modifications is simply to get the automatically collected sources through the Clang compiler without any accidents and without having to lend a hand. I hope I will find a better solution with the next OCCT version.

**Comment by borioan** [comment]
Excellent!!! I will write the first impressions of using Macad|3D in another discussion! Thank you very much!!
